### PR TITLE
Add SwiftSyntax 510.0.0

### PR DESCRIPTION
**Pull request by SimplyDanny**
https://github.com/apple/swift-syntax/releases/tag/510.0.0

### CODE REVIEW

**Comment by bazel-io** [comment]
Hello @keith, modules you maintain (swift-syntax) have been updated in this PR. Please review the changes.

File: modules/swift-syntax/510.0.0/patches/module_dot_bazel.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+ module(
+     name = "swift-syntax",
+-    version = "0",  # Update on release/* branches
++    version = "510.0.0",
+     compatibility_level = 1,
+ )
+
+-bazel_dep(name = "rules_swift", version = "1.5.1", repo_name = "build_bazel_rules_swift")
++bazel_dep(name = "rules_swift", version = "1.16.0", repo_name = "build_bazel_rules_swift")
++bazel_dep(name = "apple_support", version = "1.11.1", repo_name = "build_bazel_apple_support")
```
Comment (keith) [inline modules/swift-syntax/510.0.0/patches/module_dot_bazel.patch:13]: should i update these upstream? what was the issue without these bumps? (async question, not a huge deal)

**Review by keith — APPROVED** [review]

File: modules/swift-syntax/510.0.0/patches/module_dot_bazel.patch (hunk 1 above)
Comment (keith) [reply modules/swift-syntax/510.0.0/patches/module_dot_bazel.patch:13]: https://github.com/apple/swift-syntax/pull/2521

**Comment by keith** [comment]
thanks!

**Pull request merged**